backend/Dockerfiles/Dockerfile.engine

```diff
@@ -7,7 +8,8 @@ FROM python:3.9-bullseye AS builder
 
 # Install the Docker repository
 # hadolint ignore=DL3008
-RUN apt-get update \
+RUN --mount=type=cache,target=/var/cache/apt \
```

Comment:
Would it be a best practice to specify `sharing=locked` for the Apt cache mount, even if we don't expect the cache to ever be shared between parallel builds?

I ask because it's mentioned here: https://docs.docker.com/build/cache/optimize/#use-cache-mounts

Reply:
Good catch, yep, we definitely want to use `sharing=locked` here since long term we want to be able to parallelize our builds (like we currently do with the Java images).

Reply:
I've set `sharing=locked` on both Apt (since it's documented as necessary) and Pipenv (since I can't find any info on either way, so setting it to be safe).